---
float32 distance
float32 average_velocity
